File: app/src/androidTest/java/com/pixlee/pixleeandroidsdk/HMACTest.java
package com.pixlee.pixleeandroidsdk;

import androidx.test.runner.AndroidJUnit4;

import com.pixlee.pixleesdk.network.HMAC;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;

/**
 * Created by sungjun on 2020-02-03.
 */
@RunWith(AndroidJUnit4.class)
public class HMACTest {

    @Test
    public void success() throws JSONException, InvalidKeyException, NoSuchAlgorithmException {
        JSONObject json = new JSONObject();
        json.put("title", "a");
        json.put("email", "a@a.com");
        json.put("album_id", "4570619");


        String text = json.toString().replace("\\/", "/" );
        String test = HMAC.computeHmac(text, BuildConfig.PIXLEE_SECRET_KEY);
        System.out.println("HMACTest.text: " + text);
        System.out.println("HMACTest.result: " + test);

    }

}

File: app/src/androidTest/java/com/pixlee/pixleeandroidsdk/HMACTest.kt
package com.pixlee.pixleeandroidsdk

import com.google.gson.Gson
import com.google.gson.JsonObject
import com.pixlee.pixleesdk.network.HMAC
import org.json.JSONException
import org.junit.Assert
import org.junit.Test
import java.security.InvalidKeyException
import java.security.NoSuchAlgorithmException

/**
 * Created by sungjun on 2/1/21.
 */
class HMACTest {
    @Test
    @Throws(JSONException::class, InvalidKeyException::class, NoSuchAlgorithmException::class)
    fun `test-HMAC-SHA1`() {
        val text: String = Gson().toJson(JsonObject().apply {
            addProperty("album_id", 12345)
            addProperty("title", "Testing Photo Upload")
            addProperty("approved", true)
            addProperty("email", "test@example.com")
            addProperty("username", "Submitter Person")
            addProperty("photo_uri", "https://example.com/test.jpg")
        }).replace("\\/", "/")

        val test = HMAC.computeHmac(text, "ABCDEFG")
        val targetResult = "epBvDlHbQho/rNDdQVJowWMtGsg="
        Assert.assertEquals(targetResult, test)
    }
}